### DISCUSSION TITLE
nixos-with-flakes/modularize-the-configuration

### DISCUSSION THREAD

**Discussion by giscus** [ryan4yin/nixos-and-flakes-book#204] [Announcements]
# nixos-with-flakes/modularize-the-configuration

An unofficial and opinionated book for beginners

https://nixos-and-flakes.thiscute.world/nixos-with-flakes/modularize-the-configuration

**Comment by AyushmanOfficial** [comment]
Loved so far helped me install and understand flakes and home-manager. I was horribly confused before 😆 

Thanks for the amazing contribution ☺️

**Comment by mb1986** [comment]
I have to admit this is a very good tutorial about the modularization! I have one question though. The sentence:
> If you set a value of an option directly, it will be set with a default priority of 1000, the same as lib.mkDefault.
(https://nixos-and-flakes.thiscute.world/nixos-with-flakes/modularize-the-configuration#lib-mkoverride-lib-mkdefault-and-lib-mkforce)

seems to be wrong or I don't understand it correctly. To the best of my knowledge plain assignment has priority of [100](https://github.com/NixOS/nixpkgs/blob/master/lib/modules.nix#L1495) (and not 1000) hence it has higher priority than `mdDefault`. Please tell me if my understanding is correct :)

**Reply by ryan4yin** [reply]
Oops—thanks for catching that! I’ve fixed it in the PR: https://github.com/ryan4yin/nixos-and-flakes-book/pull/267
